### PR TITLE
feat(Execute Workflow Node): Move Type Conversion functionality to ResourceMapper

### PR DESCRIPTION
**Pull request by CharlieKolb**
## Summary

Move `ignoreTypeMismatchErrors` and `attemptToConvertTypes` to the resourceMapper. This enables us to drop type checking in both node implementations, which kicked of some included refactoring.

The ugly part here is the inclusion of "constant" `showTypeConversionOptions` field in the resourceMapper schema, which indicates to parse strings rather than accept anything [1], which I believe to be much saner behavior.
I had to introduce this property to avoid changing existing use cases of the resourceMapper. Any mapper adopting the type options added here will also adopt this stricter string conversion automatically.

One change to call out here compared to the feature branch is that we're adopting resourceMapper's behavior of silently changing `null` to `undefined` when type-checking (though the sub-workflow trigger then replaces missing/undefined fields with `null`).
Previously (on the feature branch) we carried `null` and `undefined` over explicitly to support future default values for `undefined`, whereas `null` would be kept. However I suggest we keep the resourceMappper's behavior here in an attempt to stay in line. The implementation would be trivial in case the opposite is preferred.

Also includes an unrelated notice about node version for ExecuteWorkflow V1 and V1.1.

[1] I.e. resourceMappers elsewhere would accept `7` as a string and keep it a number, whereas new mappers would turn it into `"7"` under the hood.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-3017/feature-add-type-conversion-options-to-the-resourcemapper

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue [hunk 1]
```diff
@@ -568,5 +583,50 @@ defineExpose({
...
+						emitValueChanged();
+					}
+				"
+			/>
+		</div>
 	</div>
 </template>
+
+<style module lang="scss">
+.typeConversionOptions {
```
Comment (MiloradFilipovic) [inline packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue:629]: I suggest we increase the spacing a bit both around these options and inside it:
```suggestion
.typeConversionOptions {
	display: grid;
	padding: var(--spacing-m);
	gap: var(--spacing-2xs);
}
```

**Review by MiloradFilipovic — APPROVED** [review]
Looks good on my end (apart from failing tests). Left a few minor comments

File: packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue [hunk 2]
```diff
@@ -568,5 +583,50 @@ defineExpose({
...
+			v-if="showTypeConversionOptions && state.paramValue.schema.length > 0"
+			:class="$style.typeConversionOptions"
+		>
+			<ParameterInputFull
+				:parameter="{
+					name: 'attemptToConvertTypes',
+					type: 'boolean',
+					displayName: 'Attempt to convert types',
+					default: false,
+					description: 'Attempt to convert types when mapping fields',
```
[the remark this answers is not in the document]
Comment (CharlieKolb) [reply packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue:596]: Ah thank you, I took this from the nodes where we don't have that habit. Good shout 🙌

File: packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue (hunk 1 above)
Comment (CharlieKolb) [reply packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue:629]: This is so much better 🙏

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { loadWorkflowInputMappings } from 'n8n-nodes-base/dist/utils/workflowInputsResourceMapping/GenericFunctions';
```
Comment (MiloradFilipovic) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts:1]: My main reason for keeping these separate is for us to be able to customize them for the two nodes. Now realizing that they should behave the same and there is a bit of logic there to maintain to makes sense to unify it.

**Review by MiloradFilipovic — APPROVED** [review]

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts (hunk 3 above)
Comment (CharlieKolb) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts:1]: Cool, I probably should have asked first, this just got caught up in the large merge conflict in general :D If we do modify behavior in the future it should probably still call a shared util at and of day, so let's go with this 👍

**Pull request merged**